### PR TITLE
beets: fix building

### LINKED ISSUE #268516: Build failure: beets
### Steps To Reproduce

1. build `beets`

Buiid attempted on `master`. Looks like a sphinx issue?

### Build log

https://gist.github.com/lunik1/f99cb57de250fab7a6a21a4f3b7d5bfa

### Notify maintainers

@aszlig @doronbehar @lovesegfault @pjones  

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.0, NixOS, 23.05 (Stoat), 23.05.20231118.decdf66`
 - multi-user?: `yes`
 - sandbox: `relaxed`
 - version: `nix-env (Nix) 2.18.1`
 - channels(root): `"nixos"`
 - channels(lunik1): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

### PR DESCRIPTION
**Pull request by jackwilsdon**
## Description of changes

This PR contains two changes which were required to fix `beets`:

1. e13fa51765126968c65da9368f9397dd76969543 - Sphinx 6.0.0 changed extlinks to always require placeholders in link captions. This was fixed in beets in https://github.com/beetbox/beets/commit/2106f47 but no releases contain this fix.
2. 8bdeef4f813cfc236a60b9b01456482c973f9237 - #263650 switched from using `pytest` to `pytestCheckHook`, which enabled the check hook for pytest. This started throwing errors if a plugin without any tests was disabled, as it would add the plugin's test to the disabled test list even though the test didn't exist.

The second change also fixes building of anything which depended on `beets-minimal`:

* `beetsPackages.alternatives`
* `beetsPackages.copyartifacts`
* `beetsPackages.extrafiles`

Fixes #268516.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/audio/beets/builtin-plugins.nix [hunk 1]
```diff
@@ -17,20 +17,32 @@
   };
   acousticbrainz.propagatedBuildInputs = [ python3Packages.requests ];
   albumtypes = { };
-  aura.propagatedBuildInputs = with python3Packages; [ flask pillow ];
-  badfiles.wrapperBins = [ mp3val flac ];
+  aura = {
+    propagatedBuildInputs = with python3Packages; [ flask pillow ];
+    testPaths = [ ];
```
Comment (jackwilsdon) [inline pkgs/tools/audio/beets/builtin-plugins.nix:22]: I'm not overly fond of having to list every single plugin which doesn't have tests but the only other option I could see was to modify `disabledTestPaths` in `preDist` via a shell script (removing all files which don't exist), which seemed a bit nasty.

**Review by bjornfor — APPROVED** [review]

**Comment by bjornfor** [comment]
Thanks!

**Comment by doronbehar** [comment]
Hey all, I haven't delved into the details of this PR, but I experienced an issue related to this change which maybe could be improved. I had the following of my overlays:

```nix
self: super: {
  # select plugins in beets
  beets = super.beets.override {
    pluginOverrides = {
      # Many other overrides with simple `enable = false/true` that
      # didn't cause issues
      # ...
      # This one is problematic
      sonosUpdate = { enable = false; };
    };
  };
}
```

And I got the build error:

```
Disabled tests path "test/test_sonosUpdate.py" does not exist. Aborting
```

I guessed correctly that I needed instead:

```nix
      sonosUpdate = { enable = false; testPaths = []; };
```

But it wasn't trivial to figure that out... Could we perhaps improve the override API to always set `testPaths = []` in case `enable = false;`?

**Comment by jackwilsdon** [comment]
I don't think we can just set `testPaths` to `[]` when a plugin is disabled, as plugins that actually have tests still need their tests listed in `testPaths` so we can exclude their tests when the plugin is disabled (that's all `testPaths` is used for).

I think the issue you're seeing would have actually happened before this PR too (the package would still be assuming sonosupdate has tests), except there wouldn't have been an easy workaround.

A better solution is definitely required here though.

**Pull request merged**